docs/advanced/integration_tests.md

```diff
@@ -0,0 +1,130 @@
+# Integration tests (advanced)
+
+There are several types of Gosoline integration tests, each of them meant to be used for a specific situation. All of them have the same thing in common: the Gosoline `suite` package is responsible for configuring, starting and running an application, then the tests cases are run.
+
+### resty.Client integration tests
+
+In the [Integration tests for your API](../getting_started/integration_tests.md) example, we saw the following test:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) Test_ToEuro/ /\n}/)
+```go
+func (s *ApiTestSuite) Test_ToEuro(_ suite.AppUnderTest, client *resty.Client) error {
+	var result float64
+
+	response, err := client.R().
+		SetResult(&result).
+		Execute(http.MethodGet, "/euro/10/GBP")
+
+	s.NoError(err)
+	s.Equal(http.StatusOK, response.StatusCode())
+	s.Equal(8.0, result)
+
+	return nil
+}
+```
+
+The _Test_ToEuro_ method makes a GET call to an endpoint, in order to receive back the euro exchange value for 10 BGP. Lastly, it checks if the received value is 8.0.
+
+In order to make this GET call _Test_ToEuro_ does not need to concern itself about the IP on which the exchange application is running, nor its port. All _Test_ToEuro_ needs to know is the URL path of that endpoint, as the _client_ object does the rest.
+
+This _client_ object is provided out-of-the box by Gosoline, whenever a test is defined like this:
+
+[embedmd]:# (../../examples/getting_started/integration/api_test.go /func \(s \*ApiTestSuite\) SetupSuite/ /return definer.ApiDefiner\n}/)
+```go
+func (s *ApiTestSuite) SetupSuite() []suite.Option {
```

Comment:
Why do we care about SetupSuite here? Isn't `SetupApiDefinitions` the only thing needed?